base/applications/drwtsn32/drwtsn32.cpp

```diff
@@ -0,0 +1,118 @@
+#include "precomp.h"
+#include <psapi.h>
+
+ModuleData::ModuleData(void* addr)
+{
+    BaseAddress = addr;
+    Size = 0;
+    Unloaded = false;
+}
+
+void ModuleData::Update(HANDLE hProcess)
+{
+    MODULEINFO mi = {0};
+    GetModuleInformation(hProcess, (HMODULE)BaseAddress, &mi, sizeof(mi));
+    assert(BaseAddress == mi.lpBaseOfDll);
+    Size = mi.SizeOfImage;
+
+    ModuleName.resize(MAX_PATH);
+    DWORD dwLen = GetModuleFileNameExA(hProcess, (HMODULE)BaseAddress, &ModuleName[0], ModuleName.size());
+    ModuleName.resize(dwLen);
+}
+
+
+ThreadData::ThreadData(HANDLE handle = NULL)
+    : Handle(handle)
+{
+    memset(&Context, 0, sizeof(Context));
+}
+
+void ThreadData::Update()
+{
+    Context.ContextFlags = CONTEXT_INTEGER | CONTEXT_CONTROL | CONTEXT_DEBUG_REGISTERS;
+    GetThreadContext(Handle, &Context);
+}
+
+DumpData::DumpData()
+    :ProcessID(0)
+    ,ProcessHandle(NULL)
+{
+    memset(&ExceptionInfo, 0, sizeof(ExceptionInfo));
+}
+
+
+bool UpdateFromEvent(DEBUG_EVENT& evt, DumpData& data)
+{
+    switch(evt.dwDebugEventCode)
+    {
+    case CREATE_PROCESS_DEBUG_EVENT:
+    {
+        data.ProcessPath.resize(MAX_PATH);
+        DWORD len = GetModuleFileNameExA(evt.u.CreateProcessInfo.hProcess, NULL, &data.ProcessPath[0], data.ProcessPath.size());
```

Comment:
What about using UNICODE everywhere?
But then the STL stuff should be like std::wstring & co.
Unless you want to use ATL?

Reply:
Eventually I want to print this to a file instead of stdout, and then UNICODE is not really convenient.